pkg/segment/writer/logpacker.go

```diff
@@ -51,12 +52,17 @@ func ParseRawJsonObject(currKey string, data []byte, tsKey *string,
 				return fmt.Errorf("parseRawJsonObject: arr currKey: %v, err: %v", currKey, err)
 			}
 		case jp.String:
-			valUnescaped, err := jp.Unescape(value, jsParsingStackbuf[:])
-			if err != nil {
-				return fmt.Errorf("parseRawJsonObject: failed to unescape currKey: %v, err: %v",
-					currKey, err)
+			firstBackslash := bytes.IndexByte(value, '\\')
+			if firstBackslash != -1 {
+				valUnescaped, err := jp.Unescape(value, nil)
```

Comment:
I'm not seeing how the new code will ever give a different result than the old code, and I'm not really sure why we're doing this. Maybe add a comment about the purpose of this code (i.e., what issue you're fixing). Also, are you able to add some automated test so we don't regress on this issue? It seems like a pretty subtle bug